modules/layers/src/geojson-layer/geojson-layer.ts

```diff
@@ -331,7 +341,7 @@ export default class GeoJsonLayer<
   ExtraProps = {}
 > extends CompositeLayer<Required<GeoJsonLayerProps<DataT>> & ExtraProps> {
   static layerName = 'GeoJsonLayer';
-  static defaultProps = defaultProps;
+  static defaultProps = defaultProps as any;
```

Comment:
Can we move away from `any`? 

```suggestion
  static defaultProps = defaultProps as unknown;
```

If breaking change, can we leave a comment note for v9?

Reply:
This is odd `variance` vs `contravariance` problem with static props when extending classes :/

Changing it to unknown yields:

```
Class static side 'typeof GeoJsonLayer' incorrectly extends base class static side 'typeof CompositeLayer'.
  Types of property 'defaultProps' are incompatible.
    Type 'unknown' is not assignable to type 'DefaultProps<LayerProps<any>>'
```

But, this makes me think that type of `Layer.defaultprops` is incorrect, which is another can of worms.

Reply:
Yes...

If we omit and modify types the sub class prop types won't be compatible with the super class types. 

If the subclass types are clean extensions I suspect it will work.

I think it can be done, I have a fairly similar system in luma.gl v9 and it works very cleanly. 

Base Device defines `createBuffer(): Buffer`, WebGLDevice inherits from Device and defines `createBuffer(): WebGLBuffer`, where WebGLBuffer inherits from Buffer etc. typescript carefully checks that inheritance is correct.

Reply:
It's simpler, because `WebGLBuffer` strictly extends `Buffer`.
In our case, however, `defaultProps` for `GeoJsonLayer` are only augmenting `defaultProps` from `Layer`. 

`GeoJsonLayer.defaultProps` is completely free to skip props that are defined by `Layer.defaultProps`.

(We all probably know that ... `defaultProps` are merged from all classes in hierarchy), so in each "subclass", `defaultProps` type would be `Required<_NewPropsForThisLayer> & Partial<PropsInheritedFromBaseLayer>`

I don't know how to enforce it in typesystem. I am worried it's impossible, given that:
> Static members cannot reference class type parameters (ts2302)

appears when you attempt to use type parameters in definition of `Layer.defaultProps`.

If on the other hand, each subclass `defaultProps` would "complete", so defined as 

    static defaultProps = {...BaseLayer.defaultProps, ...myNewDefaultProps }
    
then this would _truly_ extend base `defaultProps` and we could have proper typing. But that's change in API or at least "way" we do things.

Reply:
> GeoJsonLayer.defaultProps is completely free to skip props that are defined by Layer.defaultProps.

That makes no sense to me. It breaks basic OO programming rules.

> I don't know how to enforce it in typesystem. I am worried it's impossible, given that: 
> Static members cannot reference class type parameters (ts2302)

Perhaps worth checking how React does it, as this defaultProps system was modeled after React.

Looking at the overall design, I am not sure that `defaultProps` absolutely needs to be static. Perhaps it could be a method or accessor `getDefaultProps()`?. But if that is required to ensure the soundness of the type system, that would definitely be bigger v9 change.